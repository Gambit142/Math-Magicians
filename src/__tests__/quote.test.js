import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import QuotePage from '../components/quote';

describe('Check functionality of QuotePage Component', () => {
  test('Check if it renders in DOM', () => {
    render(<QuotePage />);
    expect(screen.queryByText(/Searches for JavaScript/)).toBeNull();
  });
  test('Ensure that text content not included are not rendered in the DOM', () => {
    render(<QuotePage />);
    expect(screen.queryByText(/Understanding is the basics of Mathematics - Frank Geraldine/)).toBeNull();
  });
  it('renders correctly', () => {
    const tree = renderer
      .create(<QuotePage />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});