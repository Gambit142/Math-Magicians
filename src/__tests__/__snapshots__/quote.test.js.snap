// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check functionality of QuotePage Component renders correctly 1`] = `
<div
  className="quote-div"
>
  <p>
    ‘Obvious’ is the most dangerous word in mathematics. - Eric Temple Bell
  </p>
</div>
`;
